flink-connectors/flink-connector-jdbc/src/test/java/org/apache/flink/connector/jdbc/xa/JdbcExactlyOnceSinkE2eTest.java

```diff
@@ -223,6 +244,7 @@ public void open(Configuration parameters) throws Exception {
                 runningSources =
                         new CountDownLatch(getRuntimeContext().getNumberOfParallelSubtasks());
             }
+            runningSourcesLocal = runningSources;
```

Comment:
would it possible that `runningSources` has not been defined (since it is only initialized for task0) yet before assign to `runningSourcesLocal `?

Reply:
Right, thanks! 
Fixed by adding a map from attempt number to latch (`latchesPerAttempts`).